numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,263 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
+        }
+    }
+    /* Now do the sorting */
+    sit = (PyArrayIterObject *)
+            PyArray_IterAllButAxis((PyObject *)self, &axis);
+    if (sit == NULL) {
+        goto fail;
+    }
+    if (!any_key_is_object) {
+        NPY_BEGIN_THREADS;
+    }
+    size = sit->size;
+    // Number of elements in a slice is denoted by N
+    N = PyArray_DIMS(mps[0])[axis];
+    self_stride = PyArray_STRIDE(self, axis);
+    self_elsize = PyArray_DESCR(self)->elsize;
+    maxelsize = PyArray_DESCR(mps[0])->elsize;
+    needcopy = (self_stride != self_elsize);
+    for (j = 0; j < n; j++) {
+        needcopy = needcopy
+            || PyArray_ISBYTESWAPPED(mps[j])
+            || !(PyArray_FLAGS(mps[j]) & NPY_ARRAY_ALIGNED)
+            || (PyArray_STRIDES(mps[j])[axis] != (npy_intp)PyArray_DESCR(mps[j])->elsize);
+        maxelsize = PyArray_MAX(maxelsize, PyArray_DESCR(mps[j])->elsize);
+    }
+    assert(N > 0);  /* Guaranteed and assumed by indbuffer */
+    npy_intp *indbuffer = PyDataMem_NEW(N * sizeof(*indbuffer));
+    if (indbuffer == NULL) {
+       PyErr_NoMemory();
+       goto fail;
+    }
+    if (needcopy) {
+        char *valbuffer;
+        // Create a buffer of size of the current 1D slice i.e N
+        npy_intp valbufsize = N * maxelsize;
+        if (NPY_UNLIKELY(valbufsize) == 0) {
+            valbufsize = 1;  /* Ensure allocation is not empty */
+        }
+        valbuffer = PyDataMem_NEW(valbufsize);
+        if (valbuffer == NULL) {
+            PyDataMem_FREE(indbuffer);
+            PyErr_NoMemory();
+            goto fail;
+        }
+        while (size--) {
+            for (i = 0; i < N; i++) {
+                indbuffer[i] = i;
+            }
+            for (j = 0; j < n; j++) {
+                int rcode;
+                elsize = PyArray_DESCR(mps[j])->elsize;
+                astride = PyArray_STRIDES(mps[j])[axis];
+                argsort = PyArray_DESCR(mps[j])->f->argsort[NPY_STABLESORT];
+                if(argsort == NULL) {
+                    argsort = npy_atimsort;
+                }
+                _unaligned_strided_byte_copy(valbuffer, (npy_intp) elsize,
+                                             its[j]->dataptr, astride, N, elsize);
+                if (PyArray_ISBYTESWAPPED(mps[j])) {
+                    _strided_byte_swap(valbuffer, (npy_intp) elsize, N, elsize);
+                }
+                rcode = argsort(valbuffer, indbuffer, N, mps[j]);
+                if (rcode < 0 || (PyDataType_REFCHK(PyArray_DESCR(mps[j]))
+                            && PyErr_Occurred())) {
+                    PyDataMem_FREE(valbuffer);
+                    PyDataMem_FREE(indbuffer);
+                    goto fail;
+                }
+                PyArray_ITER_NEXT(its[j]);
+            }
+            char *slicebuffer = PyDataMem_NEW(N*self_elsize);
+            _unaligned_strided_byte_copy(slicebuffer, self_elsize, sit->dataptr,
+                                         self_stride, N, self_elsize);
```

Comment:
That `_inaligned_strided_byte_copy` call is not necessary at all anymore, is it?

Reply:
It is not actually necessary, left it in for readability, moving it into the for loop below may not make the unaligned nature of the copy obvious. I am fine either way